### ISSUE TITLE
feature wishes 2022

### ISSUE THREAD

**Issue by tg123** [kubernetes-client/csharp#751]
- [ ] target netstandard2.0: lightweight client, basic model and basic apis, no websocket or api requires newer .net runtime
- [ ] remove microsoft.rest: better httpclient
- [ ] api redesign: group apis for easier access (ide autocomplete)
- [ ] kubectl experience api
- [ ] go/java client parity 
- [ ] doc server
- [ ] net5 code generator to replace generated codes
- [ ] more contributors

**Comment by goenning** [comment]
Would it be useful to have a ITokenProvider that uses Azure Identity to authenticate against AKS? 

I see there’s one for GCP, we could contribute with one for Azure, but it’d add a package reference to Azure.Identity though

**Comment by tg123** [comment]
> Would it be useful to have a ITokenProvider that uses Azure Identity to authenticate against AKS?
> 
> I see there’s one for GCP, we could contribute with one for Azure, but it’d add a package reference to Azure.Identity though

Thanks PR are welcomed. 
I am thinking to move all to KubernetesClient.TokenProvider or KubernetesClient.TokenProvider.XXX

**Comment by goenning** [comment]
Yeah, I’d prefer the TokenProvider.XXX to avoid having vendor specific dependencies all bundled in one package
